cmd/suggestion/chocolate/v1beta1/Dockerfile

```diff
@@ -21,6 +22,9 @@ RUN GRPC_HEALTH_PROBE_VERSION=v0.3.1 && \
 ADD ./pkg/ ${TARGET_DIR}/pkg/
 ADD ./${SUGGESTION_DIR}/ ${TARGET_DIR}/${SUGGESTION_DIR}/
 WORKDIR  ${TARGET_DIR}/${SUGGESTION_DIR}
+RUN if [ "$(uname -m)" = "aarch64" ]; then \
+    sed -i -e '$a git+https://github.com/fmder/ghalton@master' -e '/^ghalton/d' requirements.txt; \
```

Comment:
Since `ghalton` for `aarch64` is not distributed in the `PyPI` [repository](https://pypi.org/project/ghalton/#files), `ghalton` will be built for `aarch64` when we build the container image although we can not build `ghalton` for `aarch64` because the following errors occur.

So we need to build it from the latest source code.

```shell
$ docker build --platform linux/arm64 -f cmd/suggestion/chocolate/v1beta1/Dockerfile .
...
[+] Building 302.8s (11/12)                                                                                                                                                                                                             
 => ERROR [7/8] RUN pip install --no-cache-dir -r requirements.txt                                                                                                                                                               261.2s
------
 > [7/8] RUN pip install --no-cache-dir -r requirements.txt:
#11 17.24 Collecting git+https://github.com/AIworx-Labs/chocolate@master (from -r requirements.txt (line 10))
#11 17.25   Cloning https://github.com/AIworx-Labs/chocolate (to revision master) to /tmp/pip-req-build-o3m36yqu
#11 17.25   Running command git clone -q https://github.com/AIworx-Labs/chocolate /tmp/pip-req-build-o3m36yqu
#11 19.65   Resolved https://github.com/AIworx-Labs/chocolate to commit 0ba4f6f0130eab851d32d5534241c8cac3f6666e
#11 35.70 Collecting grpcio==1.41.1
#11 35.89   Downloading grpcio-1.41.1-cp39-cp39-manylinux_2_17_aarch64.whl (44.2 MB)
#11 41.66 Collecting cloudpickle==0.5.6
#11 41.69   Downloading cloudpickle-0.5.6-py2.py3-none-any.whl (16 kB)
#11 48.34 Collecting numpy>=1.20.0
#11 48.38   Downloading numpy-1.21.4-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (13.0 MB)
#11 53.32 Collecting scikit-learn>=0.24.0
#11 53.35   Downloading scikit_learn-1.0.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (26.4 MB)
#11 60.44 Collecting scipy>=1.5.4
#11 60.47   Downloading scipy-1.7.2-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (36.6 MB)
#11 65.40 Collecting forestci==0.3
#11 70.08   Downloading forestci-0.3-py3-none-any.whl (11 kB)
#11 73.64 Collecting protobuf==3.19.1
#11 73.68   Downloading protobuf-3.19.1-cp39-cp39-manylinux2014_aarch64.whl (913 kB)
#11 74.01 Collecting googleapis-common-protos==1.6.0
#11 74.04   Downloading googleapis-common-protos-1.6.0.tar.gz (35 kB)
#11 81.57 Collecting SQLAlchemy==1.4.26
#11 81.71   Downloading SQLAlchemy-1.4.26-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (1.6 MB)
#11 82.27 Collecting ghalton>=0.6.2
#11 82.31   Downloading ghalton-0.6.2.tar.gz (124 kB)
#11 89.97 Collecting cython>=0.29.24
#11 90.01   Downloading Cython-0.29.24-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (1.9 MB)
#11 90.66 Collecting six>=1.5.2
#11 90.70   Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
#11 91.08 Collecting nose>=1.1.2
#11 91.11   Downloading nose-1.3.7-py3-none-any.whl (154 kB)
#11 94.02 Collecting greenlet!=0.4.17
#11 94.07   Downloading greenlet-1.1.2-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (154 kB)
#11 95.63 Collecting joblib>=0.11
#11 95.66   Downloading joblib-1.1.0-py2.py3-none-any.whl (306 kB)
#11 95.88 Collecting threadpoolctl>=2.0.0
#11 95.91   Downloading threadpoolctl-3.0.0-py3-none-any.whl (14 kB)
#11 100.0 Collecting pandas>=0.19
#11 100.1   Downloading pandas-1.3.4-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (10.9 MB)
#11 102.0 Collecting dataset>=0.8
#11 102.0   Downloading dataset-1.5.1-py2.py3-none-any.whl (18 kB)
#11 102.3 Collecting filelock>=2.0
#11 102.4   Downloading filelock-3.4.0-py3-none-any.whl (9.8 kB)
#11 103.0 Collecting banal>=1.0.1
#11 103.0   Downloading banal-1.0.6-py2.py3-none-any.whl (6.1 kB)
#11 103.7 Collecting alembic>=0.6.2
#11 103.8   Downloading alembic-1.7.5-py3-none-any.whl (209 kB)
#11 104.3 Collecting Mako
#11 104.3   Downloading Mako-1.1.6-py2.py3-none-any.whl (75 kB)
#11 106.6 Collecting pytz>=2017.3
#11 106.7   Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)
#11 107.1 Collecting python-dateutil>=2.7.3
#11 107.1   Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
#11 108.5 Collecting MarkupSafe>=0.9.2
#11 108.5   Downloading MarkupSafe-2.0.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (26 kB)
#11 108.7 Building wheels for collected packages: googleapis-common-protos, chocolate, ghalton
#11 108.7   Building wheel for googleapis-common-protos (setup.py): started
#11 112.0   Building wheel for googleapis-common-protos (setup.py): finished with status 'done'
#11 112.0   Created wheel for googleapis-common-protos: filename=googleapis_common_protos-1.6.0-py3-none-any.whl size=77592 sha256=bcbbf5e5f98655b20578874040039b019e6d806cfdd93dae37c7b3a358018687
#11 112.0   Stored in directory: /tmp/pip-ephem-wheel-cache-hxszj4s1/wheels/e8/1f/11/8233e1627c658357f6be4ea4e353e54034485fc9496afc333f
#11 112.0   Building wheel for chocolate (setup.py): started
#11 124.3   Building wheel for chocolate (setup.py): finished with status 'done'
#11 124.3   Created wheel for chocolate: filename=chocolate-0.6-cp39-cp39-linux_aarch64.whl size=79025 sha256=7f75cd4f23e278d6134c15ec0e883986925ae3faffebe3cf5087368aae8b140b
#11 124.3   Stored in directory: /tmp/pip-ephem-wheel-cache-hxszj4s1/wheels/68/33/a2/82eeb15c0669e449fb81d64e03ebb1f04a97338857164087d1
#11 124.3   Building wheel for ghalton (setup.py): started
#11 126.9   Building wheel for ghalton (setup.py): finished with status 'error'
#11 126.9   ERROR: Command errored out with exit status 1:
#11 126.9    command: /usr/local/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/setup.py'"'"'; __file__='"'"'/tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-n8e3ml0b
#11 126.9        cwd: /tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/
#11 126.9   Complete output (16 lines):
#11 126.9   running bdist_wheel
#11 126.9   running build
#11 126.9   running build_py
#11 126.9   creating build
#11 126.9   creating build/lib.linux-aarch64-3.9
#11 126.9   creating build/lib.linux-aarch64-3.9/ghalton
#11 126.9   copying ghalton/__init__.py -> build/lib.linux-aarch64-3.9/ghalton
#11 126.9   copying ghalton/ghalton_wrapper.py -> build/lib.linux-aarch64-3.9/ghalton
#11 126.9   copying ghalton/constants.py -> build/lib.linux-aarch64-3.9/ghalton
#11 126.9   running build_ext
#11 126.9   building 'ghalton._ghalton_wrapper' extension
#11 126.9   creating build/temp.linux-aarch64-3.9
#11 126.9   creating build/temp.linux-aarch64-3.9/src
#11 126.9   gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include/python3.9 -c src/Halton.cpp -o build/temp.linux-aarch64-3.9/src/Halton.o -stdlib=libc++
#11 126.9   gcc: error: unrecognized command-line option ‘-stdlib=libc++’
#11 126.9   error: command '/usr/bin/gcc' failed with exit code 1
#11 126.9   ----------------------------------------
#11 126.9   ERROR: Failed building wheel for ghalton
#11 126.9   Running setup.py clean for ghalton
#11 129.0 Successfully built googleapis-common-protos chocolate
#11 129.0 Failed to build ghalton
#11 132.2 Installing collected packages: MarkupSafe, greenlet, SQLAlchemy, six, numpy, Mako, threadpoolctl, scipy, pytz, python-dateutil, joblib, banal, alembic, scikit-learn, protobuf, pandas, nose, filelock, dataset, grpcio, googleapis-common-protos, ghalton, forestci, cython, cloudpickle, chocolate
#11 256.9     Running setup.py install for ghalton: started
#11 259.1     Running setup.py install for ghalton: finished with status 'error'
#11 259.1     ERROR: Command errored out with exit status 1:
#11 259.1      command: /usr/local/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/setup.py'"'"'; __file__='"'"'/tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-a54tcg7t/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.9/ghalton
#11 259.1          cwd: /tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/
#11 259.1     Complete output (16 lines):
#11 259.1     running install
#11 259.1     running build
#11 259.1     running build_py
#11 259.1     creating build
#11 259.1     creating build/lib.linux-aarch64-3.9
#11 259.1     creating build/lib.linux-aarch64-3.9/ghalton
#11 259.1     copying ghalton/__init__.py -> build/lib.linux-aarch64-3.9/ghalton
#11 259.1     copying ghalton/ghalton_wrapper.py -> build/lib.linux-aarch64-3.9/ghalton
#11 259.1     copying ghalton/constants.py -> build/lib.linux-aarch64-3.9/ghalton
#11 259.1     running build_ext
#11 259.1     building 'ghalton._ghalton_wrapper' extension
#11 259.1     creating build/temp.linux-aarch64-3.9
#11 259.1     creating build/temp.linux-aarch64-3.9/src
#11 259.1     gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include/python3.9 -c src/Halton.cpp -o build/temp.linux-aarch64-3.9/src/Halton.o -stdlib=libc++
#11 259.1     gcc: error: unrecognized command-line option ‘-stdlib=libc++’
#11 259.1     error: command '/usr/bin/gcc' failed with exit code 1
#11 259.1     ----------------------------------------
#11 259.1 ERROR: Command errored out with exit status 1: /usr/local/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/setup.py'"'"'; __file__='"'"'/tmp/pip-install-qproh29v/ghalton_0d24e7cd43784905ac5436e2a6fa9f90/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-a54tcg7t/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.9/ghalton Check the logs for full command output.
#11 260.2 WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
#11 260.2 You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
------
executor failed running [/bin/sh -c pip install --no-cache-dir -r requirements.txt]: exit code: 1
```